### ISSUE TITLE
Room - Removed room member reappears as a member of the room when navigate from Search

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#35289]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.32-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Login as User A
2. Navigate to Settings> Workspace> New workspace> Invite User B
3. User A creates Public room and invite User B
4. User A removes User B from room
5. As User B via search open the room

## Expected Result:
User B should not be the member of the room since he is removed

## Actual Result:
Removed room member User B reappears as a member of the room when navigate from Search

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/cf22bccd-5156-4184-b12e-eef7b4b92bb1

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave8. 
CC @zanyrenney

**Comment by JordanLevy19** [comment]
Unable to replicate on the specified supported platforms - however the search was still a bit buggy while testing so suspect there may be some clues there - 


**Please re-state the problem that we are trying to solve in this issue.**
There's a bug where a user who has been removed from a chat room can still access, and interact with that room through the search function. 

**Root Cause of the Problem:**
The root cause of the issue, where a recently removed user still sees a chat room in search results, may lie in the delayed or inconsistent updating of the BaseOptionsSelector component's state. 

**Suggested Changes to Solve the Problem:**

**Ensure Real-Time Data Updates:**
Modify the componentDidUpdate method to more effectively handle updates related to room membership changes. This ensures that any changes in the user's status are immediately reflected in the search results.

componentDidUpdate(prevProps) {
    if (!_.isEqual(this.props.sections, prevProps.sections)) {
        const newSections = this.sliceSections();
        this.setState({ sections: newSections });
    }
    ...
}

**Refine Search Result Filtering:**
Adjust the logic in methods like sliceSections or flattenSections to filter out rooms that the user should no longer have access to. This may involve adding checks to confirm the user's current membership status before including a room in the list.

sliceSections() {
    return _.map(this.props.sections, (section) => {
        if (_.isEmpty(section.data) || !userHasAccess(section)) {
            return section;
        }
        ...
    });
}

**Optimize State Management in Search:**
In the updateSearchValue method, ensure that the search results are not just based on the input value but also consider the user's current room memberships.

updateSearchValue(value) {
    this.setState({
        paginationPage: 1,
        value,
    });
    this.props.onChangeText(value);
    this.refreshSearchResults(); // Implement this method to refresh the list based on current membership status
}

**Comment by abdulrahuman5196** [comment]
Will work on review today.

**Comment by abdulrahuman5196** [comment]
@lanitochka17 In the video, at last user B has messaged `hi` and joined the chat, since its a public room it adds them as the member right? Isn't this expected.
@abekkala

**Comment by abekkala** [comment]
> messaged hi and joined the chat, since its a public room it adds them as the member right? Isn't this expected.

yes, this is expected. if you access a room and take an action such as leaving a comment, you are added as a member.

**Comment by abdulrahuman5196** [comment]
@abekkala Then could we close this issue as expected?
